services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/IndexingController.java

```diff
@@ -48,35 +48,52 @@
     private static final Logger log = LoggerFactory.getLogger(IndexingController.class);
 
     @Autowired
-    private IndexingService indexingService;
+    private IndexingService solrIndexingService;
+
+    @Autowired
+    private IndexingService triplesIndexingService;
 
     /**
-     * Perform a deep reindexing operation on the object with the specified id and all of its children.
+     * Perform a deep reindexing operation in solr on the object with the
+     * specified id and all of its children.
      *
      * @param id the identifier of the object to be reindexed
-     * @param inplace whether the reindex should be an in-place recursive reindex (optional)
+     * @param inplace whether the reindex should be an in-place recursive
+     *            reindex (optional)
      * @return
      */
     @RequestMapping(value = "edit/solr/reindex/{id}", method = RequestMethod.POST)
     public ResponseEntity<Object> reindex(@PathVariable("id") String id,
             @RequestParam(value = "inplace", required = false) Boolean inplace) {
-        return indexObjectAndChildren(id, inplace);
+        return indexObjectAndChildren(id, inplace, solrIndexingService, "reindexSolr");
     }
 
     /**
-     * Perform a shallow reindexing of the object specified by id
+     * Perform a shallow reindexing in solr of the object specified by id
      *
      * @param id the identifier of the object to be reindexed
      * @return
      */
     @RequestMapping(value = "edit/solr/update/{id}", method = RequestMethod.POST)
     public ResponseEntity<Object> reindex(@PathVariable("id") String id) {
-        return indexObject(id);
+        return indexObject(id, solrIndexingService, "updateSolr");
+    }
+
+    /**
+     * Perform a deep reindexing operation in the triple store on the object
+     * with the specified id.
+     *
+     * @param id the identifier of the object to be reindexed
+     * @return
+     */
+    @RequestMapping(value = "edit/triples/reindex/{id}", method = RequestMethod.POST)
+    public ResponseEntity<Object> reindexTriples(@PathVariable("id") String id) {
+        return indexObjectAndChildren(id, false, triplesIndexingService, "reindexTriples");
     }
 
-    private ResponseEntity<Object> indexObject(String id) {
+    private ResponseEntity<Object> indexObject(String id, IndexingService indexingService, String action) {
```

Comment:
why is "action" being passed in as a param here? wouldn't the action be "reindex triplestore"?

Reply:
Yes, the controller methods pass in different values depending on what is being indexed (`reindexSolr`, `reindexTriples`, `updateSolr`), but share the same methods for kicking off the indexing request. Its a bit unnecessary in this case because there isn't an endpoint for indexing a single object in the triple store, I just added it for consistency so that the controller method was where the action name was always coming from. I can remove it if you'd prefer though.